programs/disks/DisksApp.cpp

```diff
@@ -202,59 +449,44 @@ int DisksApp::main(const std::vector<String> & /*args*/)
+    String main_disk = config().getString("disk", "default");
 
-    auto command_options = command->getCommandOptions();
-    std::vector<String> args;
-    if (command_options)
-    {
-        auto parser = po::command_line_parser(command_arguments).options(*command_options).allow_unregistered();
-        po::parsed_options parsed = parser.run();
-        po::store(parsed, options);
-        po::notify(options);
+    auto validator = [](const Poco::Util::AbstractConfiguration &, const std::string &, const std::string &) { return true; };
 
-        args = po::collect_unrecognized(parsed.options, po::collect_unrecognized_mode::include_positional);
-        command->processOptions(config(), options);
-    }
-    else
-    {
-        auto parser = po::command_line_parser(command_arguments).options({}).allow_unregistered();
-        po::parsed_options parsed = parser.run();
-        args = po::collect_unrecognized(parsed.options, po::collect_unrecognized_mode::include_positional);
-    }
+    constexpr auto config_prefix = "storage_configuration.disks";
+    auto disk_selector = std::make_shared<DiskSelector>();
+    disk_selector->initialize(config(), config_prefix, global_context, validator);
 
-    std::unordered_set<std::string> disks
-    {
-        config().getString("disk", "default"),
-        config().getString("disk-from", config().getString("disk", "default")),
-        config().getString("disk-to", config().getString("disk", "default")),
-    };
-
-    auto validator = [&disks](
-        const Poco::Util::AbstractConfiguration & config,
-        const std::string & disk_config_prefix,
-        const std::string & disk_name)
-    {
-        if (!disks.contains(disk_name))
-            return false;
+    std::vector<std::pair<DiskPtr, std::optional<String>>> disks_with_path;
 
-        const auto disk_type = config.getString(disk_config_prefix + ".type", "local");
+    for (const auto & [_, disk_ptr] : disk_selector->getDisksMap())
+    {
+        disks_with_path.emplace_back(
+            disk_ptr, (disk_ptr->getName() == "local") ? std::optional{fs::current_path().string()} : std::nullopt);
+    }
 
-        if (disk_type == "cache")
-            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Disk type 'cache' of disk {} is not supported by clickhouse-disks", disk_name);
 
-        return true;
-    };
+    client = std::make_unique<DisksClient>(std::move(disks_with_path), main_disk);
 
-    constexpr auto config_prefix = "storage_configuration.disks";
-    auto disk_selector = std::make_shared<DiskSelector>();
-    disk_selector->initialize(config(), config_prefix, global_context, validator);
+    suggest.setCompletionsCallback([&](const String & prefix, size_t /* prefix_length */) { return getCompletions(prefix); });
 
-    command->execute(args, disk_selector, config());
+    if (!query.has_value())
+    {
+        runInteractiveReplxx();
+    }
+    else
+    {
+        processQueryText(query.value());
```

Comment:
Is it possible to execute commands sequentially?

Reply:
Do you mean several commands in one query? Right now it its impossible